### PR TITLE
Aks cert manager

### PR DESCRIPTION
**Pull request by jrudley**
In a comment on the pull request, include the following information and check off the list.

 - [ ] Mini bio
 - [ ] Desired date of publishing
  - 07-20-2018
 - [ ] Social media contact info
 - [ ] Redaction: (yes|no) YES
  - Allow redaction, editing or structure changes by technical writer staff
 - [ ] Comments: (yes|no) YES
  - If you're willing to answer comments about your post, set `comments: true`.
  - Make sure to subscribe to your post comments by clicking on the **Subscribe** link at the very bottom of the post _once_ it's published.

@catlook Please review.

### CODE REVIEW

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-ca6ef36afc/)

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-9cb6dc012c/)

File: _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+date: 2018-07-20 00:00
+comments: true
+author: Jimmy Rudley
+published: true
+authorIsRacker: true
+categories:
+    - General
+---
+
+Azure has recently went GA with their Azure Kubernetes Service, but following the Microsoft documentation for setting up an HTTPS ingress controller could make you go in circles. Let me share what I have found and what Microsoft recently changed in their HTTPS ingress controller [document](https://docs.microsoft.com/en-us/azure/aks/ingress)
```
Comment (catlook) [inline _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md]: Azure recently ...
... found about Microsoft's recent changes in their ...
[Ed note: also, add a period after the link]

File: _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+<!-- more -->
+
+Microsoft is really putting the spotlight on their Kubernetes offering, so I spent some time setting it up. It truly is simple to deploy and manage. Microsoft has also done great work integrating it into their existing tooling set. If you have not used their cloud shell, please try it out. It has the majority of tooling already installed to manage your AKS cluster. I do want to focus around the HTTPS ingress controller to save some people the headaches I ran into.
```
Comment (catlook) [inline _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md]: ... focus on the ...

File: _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+With the Microsoft documentation, they use [Cert-Manager](https://github.com/jetstack/cert-manager) to retrieve and configure certificates for HTTPS. Cert-Manager makes use of Let's Encrypt to create the certificates. If you read the Cert-Manager readme, under current status it mentions it is not ready for a critical production stack. OK, fair enough, they warn us, but Microsoft does not. As I am writing this article, Microsoft changed their document to make use of Let's Encrypt staging environment. Previously, they were using the production environment setting. I am sure the author of their document missed this tidbit, but Let's Encrypt will rate limit your requests, especially when using the production enviornment setting. In Microsoft's example they are using the domain **demo-aks-ingress.eastus.cloudapp.azure.com**, but when I initially tested it out using **demo-jimmy-ingress.eastus.cloudapp.azure.com** I noticed my certificate never provisioned. I did a **kubectl describe certificate** and was presented with the following:
```
Comment (catlook) [inline _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md]: The Microsoft documentation uses[Cert Manager] ...
... makes use of the ``Let's Encrypt`` tool to ...
... make use of the ``Let's Encrypt`` staging ...
... but ``Let's Encrypt`` rate limits your ...
... example, they are using ...
... using **demo-jimmy-ingress.eastus.cloudapp.azure.com**, I noticed ...

File: _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+```
+Message:               Failed to finalize order: acme: urn:ietf:params:acme:error:rateLimited: Error finalizing order :: too many certificates already issued for exact set of domains: demo-jimmy-ingress.eastus.cloudapp.azure.com: see https://letsencrypt.org/docs/rate-limits/
+```
+
+I am puzzled, I was being rate limited for a domain I just submitted a request for? It turns out that Let's Encrypt will look at **Azure.com**. Here is the snippet from the rate limit document
```
Comment (catlook) [inline _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md]: ... puzzled. I was ...
... that ``Let's Encrypt`` looks at ...
... limit document:

File: _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+```
+The main limit is Certificates per Registered Domain, (20 per week). 
+A registered domain is, generally speaking, the part of the domain you purchased from your domain name registrar. 
+For instance, in the name www.example.com, the registered domain is example.com. In new.blog.example.co.uk, the registered domain is example.co.uk. We use the Public Suffix List to calculate the registered domain.
+```
+
+Today, Microsoft changed their document to use the staging environment which has these limits:
```
Comment (catlook) [inline _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md]: ... environment, which  ...

File: _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "AKS - Cert Manager Certificate Rate Limit"
+date: 2018-07-20 00:00
```
Comment (catlook) [inline _posts/2018-07-20-AKS-Cert-Manager-Lets-Encrypt.md]: change to today's date!

**Comment by rackernexus** [comment]
:eyes: [Preview](https://staging.developer.rackspace.com/build-93dba3ef9a/)

**Pull request merged**